Content.Shared/Movement/Systems/SharedMoverController.cs

```diff
@@ -567,7 +567,7 @@ private bool TryGetFootstepSound(
 
         // If the coordinates have a FootstepModifier component
         // i.e. component that emit sound on footsteps emit that sound
-        var anchored = grid.GetAnchoredEntitiesEnumerator(position);
+        var anchored = _mapSystem.GetAnchoredEntitiesEnumerator(uid, grid, position);
```

Comment:
This should be the uid of the grid, not the mover. The value isn't actually used, so it works like this anyway, but the right value should be used.
```suggestion
        var anchored = _mapSystem.GetAnchoredEntitiesEnumerator(xform.GridUid.Value, grid, position);
```